### ISSUE TITLE
Issues with accessibility when using high contrast mode in Windows

### ISSUE THREAD

**Issue by mabryl** [ckeditor/ckeditor5#14907]
## 📝 Provide detailed reproduction steps (if any)

1. While using Windows turn on the high contrast mode in the OS's settings
2. Open the [Font family, size, and color demo](https://ckeditor.com/docs/ckeditor5/latest/features/font.html#demo)
3. Open the font color/background color dropdown

### ✔️ Expected result

You should be able to see the different colors for fonts and backgounds in the text as well as in the dropdown.

### ❌ Actual result

All the colors get dropped. Here's a screenshot from Win 11 using the "Night Sky" high contrast preset:

![nightsky](https://github.com/ckeditor/ckeditor5/assets/72079603/e2392a3d-a547-48c1-a69d-16fbaa1b4ea1)

And here's one using the "Desert" preset:

![desert](https://github.com/ckeditor/ckeditor5/assets/72079603/c6d10d41-16cc-4e18-8cd8-23313b8f1e8e)

One thing to note here is that Tiny MCE also produces the same effect on Windows, so it's possible that it's some upstream issue related to the high contrast mode itself.

Also, macOS's "Invert colours" accessibility option doesn't have this issue.

## 📃 Other details

* Browser: cross-browser
* OS: Windows 10, Windows 11
* First affected CKEditor version: N/A
* Installed CKEditor plugins: N/A

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by mgifford** [comment]
There are still a lot of Windows users, and a lot of folks use their High Contrast Mode, so we really should be doing better to support it.  
  
Here is a related issue from Drupal https://www.drupal.org/project/drupal/issues/3270130

**Comment by mabryl** [comment]
Another issue is that the caret is not visible in the editable when there's no content in it:

![Screenshot 2024-04-22 101009](https://github.com/ckeditor/ckeditor5/assets/72079603/e11cd569-4206-44be-a37c-7f02d51e84d1)

You can see the caret only when you type something:

![Screenshot 2024-04-22 101114](https://github.com/ckeditor/ckeditor5/assets/72079603/96466a97-5272-43d7-af09-ad13a43bff21)
